tests/test_pydomainextractor.py

```diff
@@ -472,3 +472,569 @@ def test_load_called_with_data(
+                'subdomain': 'www',
+                'domain': 'cgs',
+                'suffix': 'act.edu.au',
+            },
+        )
+        url = 'http://www.google.com.au/'
+        self.assertEqual(
+            first=self.domain_extractor.extract_from_url(
+                url,
+            ),
+            second={
+                'subdomain': 'www',
+                'domain': 'google',
+                'suffix': 'com.au',
+            },
+        )
+
+    def test_tld_is_a_website_too(
+        self,
+    ):
+        url = 'http://www.metp.net.cn'
+        self.assertEqual(
+            first=self.domain_extractor.extract_from_url(
+                url,
+            ),
+            second={
+                'subdomain': 'www',
+                'domain': 'metp',
+                'suffix': 'net.cn',
+            },
+        )
+
+    def test_dns_root_label_raises_value_error(
+        self,
+    ):
+        url = 'http://www.example.com./'
+        with self.assertRaises(
+            ValueError,
+        ):
+            self.domain_extractor.extract_from_url(
+                url,
+            )
+
+    def test_psl_private_domains(
+        self,
+    ):
+        url = 'http://waiterrant.blogspot.com'
+        self.assertEqual(
+            first=self.domain_extractor.extract_from_url(
+                url,
+            ),
+            second={
+                'subdomain': '',
+                'domain': 'waiterrant',
+                'suffix': 'blogspot.com',
+            },
+        )
+
+    def test_ipv4(
+        self,
+    ):
+        url = 'http://127.0.0.1/foo/bar'
+        self.assertEqual(
+            first=self.domain_extractor.extract_from_url(
+                url,
+            ),
+            second={
+                'subdomain': '127.0.0',
+                'domain': '1',
+                'suffix': '',
+            },
+        )
+
+    def test_ipv4_bad(
+        self,
+    ):
+        url = 'http://256.256.256.256/foo/bar'
+        self.assertEqual(
+            first=self.domain_extractor.extract_from_url(
+                url,
+            ),
+            second={
+                'subdomain': '256.256.256',
+                'domain': '256',
+                'suffix': '',
+            },
+        )
+
+    def test_ipv4_lookalike(
+        self,
+    ):
+        url = 'http://127.0.0.1.9/foo/bar'
+        self.assertEqual(
+            first=self.domain_extractor.extract_from_url(
+                url,
+            ),
+            second={
+                'subdomain': '127.0.0.1',
+                'domain': '9',
+                'suffix': '',
+            },
+        )
+
+    def test_extract_url(
+        self,
+    ):
+        result = self.domain_extractor.extract_from_url(
+            'http://admin:password1@www.google.com:666/secret/admin/interface?param1=42'
+        )
+        assert result == {
+            'subdomain': 'www',
+            'domain': 'google',
+            'suffix': 'com',
+        }
```

Comment:
please put all these tests in their appropriate function names. Take domain test and put them inside the test case of the domains, leave URLs in one function

Reply:
I didn't get you, all url extraction functions are under TLDExtractTestCase.
Do you mean you want to see one function under the ExtractionTestCase, that contains all the url tests?
`
class DomainExtractorExtractionTestCase:
    def test_extract_from_url():
        # test with protocol
        # test with protocol and query fragments
       ....
`